src/components/TradeWidget/useNumberInput.tsx

```diff
@@ -0,0 +1,66 @@
+import { useCallback, useEffect } from 'react'
+import { useFormContext } from 'react-hook-form'
+import { leadingAndTrailingZeros, adjustPrecision, trailingZerosAfterDot, validInputPattern } from 'utils'
+import { TradeFormData } from '.'
+
+interface Params {
+  inputId: string
+  precision: number
+}
+
+interface Result {
+  removeExcessZeros: (event: React.SyntheticEvent<HTMLInputElement, Event>) => void
+  enforcePrecision: () => void
+  onKeyPress: (event: React.KeyboardEvent<HTMLInputElement>) => void
+}
+
+function _preventInvalidChars(event: React.KeyboardEvent<HTMLInputElement>): void {
+  if (!validInputPattern.test(event.currentTarget.value + event.key)) {
+    event.preventDefault()
+  }
+}
+
+export function useNumberInput(params: Params): Result {
+  const { inputId, precision } = params
+  const { setValue, watch } = useFormContext<TradeFormData>()
+  const inputValue = watch(inputId) as string
+
+  const enforcePrecision = useCallback(() => {
+    const newValue = adjustPrecision(inputValue, precision)
+    if (inputValue !== newValue) {
+      setValue(inputId, newValue, true)
+    }
+  }, [inputId, inputValue, precision, setValue])
+
+  useEffect(() => {
+    enforcePrecision()
+  }, [enforcePrecision])
```

Comment:
An almost empty hook, funny :smile:

Reply:
Does it have an issue? 

I think we can change it to:
`useEffect(enforcePrecision, [enforcePrecision])`

Doing the change now. Let me know if it's fine :)

Reply:
:+1: